app/services/user.py

```diff
@@ -142,6 +143,30 @@ def update(self) -> User:
         return self.user
 
 
+class UserDeletionService:
+    def __init__(self, user, **attrs):
+        self.user = user
+        self.scopes = attrs.pop("scopes")
+        authentication.validate_scope_permissions("delete:user", self.scopes, attrs)
+        self.attrs = attrs
+
+    def delete(self):
+        for attr, val in self.attrs.items():
+            setattr(self.user, attr, val)
+
+        if "delete:user" not in self.scopes:
+            raise AccessDenied("Insufficient permissions")
+
+        tunnels = Tunnel.query.filter_by(user=self.user)
+        for tunnel in tunnels:
+            TunnelDeletionService(self.user, tunnel).delete()
```

Comment:
If you're going to explicitly delete these, then you probably don't want the cascade deletion in the schema file.  I'm guessing these are deleted by this loop to ensure the nomad tunnels are shut down?

Reply:
Yes you're correct